### PR TITLE
feat: agent-autonomy workflow, .claude layout move, and config cleanup

### PR DESCRIPTION
**Pull request by bertini36**
## Description

Restructures the Claude Code configuration for agent-autonomous development and cleans up the repo layout. Companion to bertini36/sinsesgo#170, which adds the sinsesgo-specific skills.

### Agent-autonomy rules and workflow

- **`rules/performance.md`** (new): N+1 prevention at the queryset origin, database-side aggregation, batch writes, query-count guards in tests, no premature caching. Django ORM guidance scoped, with a stack-agnostic path for other ORMs.
- **`rules/tests.md`** (extended): "One behavior, one level". Unit covers logic, integration covers only unreachable wiring, evals cover quality; duplicates at higher levels get deleted.
- **`skills/fix-until-green`** (new): verification loop. Runs `make check` + `pre-commit`, dispatches a fixer subagent per failure, caps at 5 iterations, reports honestly on non-convergence. No assertion weakening, no test deletion to force green.
- **`skills/start-feature`**: Implement stage now defaults to `superpowers:subagent-driven-development` for plans with 3+ independent tasks; Verify stage runs `fix-until-green` on failing checks.

### Layout: `ai/claude/` moved to `.claude/`

Claude Code config now lives at the conventional `.claude/` path. README symlink instructions and prose, gitignore entries, and hook paths in `settings.json` updated; local `~/.claude` symlinks recreated and verified (two dead relic links removed: `~/.claude/commands`, `~/.claude-auto-retry.json`).

### Removals

- **`visual-plan` and `visual-recap`** skills deleted along with the `AGENT_NATIVE_PLANS_MODE` setting and their README docs (~2.1k lines).

### Docs and consistency

- README tables now document `fix-until-green`, `investigate-sentry`, and the `performance` rule; dead `agents/skills-workspace/` gitignore path corrected.
- Copilot review feedback applied: fixed a nonexistent skill reference, the unrecoverable-ledger claim, and markdown style in the new tests section.

### Granola

- `cask "granola"` added to the Brewfile with a README row.
- Official MCP server documented next to the existing `datadog-mcp` pattern: `claude mcp add --transport http --scope user granola https://mcp.granola.ai/mcp` (OAuth on first connection, no API key).

## Type of Change

- [x] Documentation / tooling
- [x] Refactor (config layout move)
- [ ] New feature
- [ ] Bug fix

## Test Procedure

- All `~/.claude` symlinks re-pointed to `.dotfiles/.claude/` and verified resolving on this machine; broken-link scan of `~` and `~/.claude` clean.
- `settings.json` validated as JSON after each edit; hook paths confirmed to exist.
- Skills, rules, and agents load from the new location in a live session.
- No pre-commit configured in this repo.

### CODE REVIEW

File: ai/claude/skills/start-feature/SKILL.md [hunk 1]
```diff
@@ -32,9 +32,9 @@ Then dispatch the `plan-evaluator` agent. With fresh context that has no stake i
...
 
-For plans made of independent tasks, `superpowers:subagent-driven-development` is an alternative: each task goes to a fresh implementer subagent, and a per-task reviewer checks the work before moving on. Its scratch files (task briefs, reports, progress ledger) live in a git-ignored `.superpowers/sdd/` directory; `git clean -fdx` deletes the progress ledger, so recover from `git log` if that happens.
+When the plan has 3 or more independent tasks, implement with `superpowers:subagent-driven-development` (preferred): each task goes to a fresh implementer subagent, and a per-task reviewer checks the work before moving on. Each subagent brief names the domain skills relevant to the files it touches (for example `django-patterns`, `sinsesgo-agent-pattern`). Its scratch files (task briefs, reports, progress ledger) live in a git-ignored `.superpowers/sdd/` directory; `git clean -fdx` deletes the progress ledger, so recover from `git log` if that happens.
```
[the remark this answers is not in the document]
Comment (bertini36) [reply ai/claude/skills/start-feature/SKILL.md]: Fixed in bc3a962: swapped `sinsesgo-agent-pattern` for `python-code-style`, which actually exists as a skill in this repo, and reworded the git-clean note since `.superpowers/sdd/` has its own `.gitignore` excluding everything (never tracked, so `git log` can't recover it).

File: .claude/rules/tests.md [hunk 2]
```diff
@@ -50,4 +50,13 @@ no explanatory comment: `test_renders_markdown_when_answer_present`,
 
 Add a comment only when the logic is genuinely non-obvious — a subtle edge case or a
 non-intuitive expected value. Descriptive names and the blank-line structure carry the
-rest. No comments that restate the code, and no phase labels.
\ No newline at end of file
+rest. No comments that restate the code, and no phase labels.
+## One behavior, one level
```
[the remark this answers is not in the document]
Comment (bertini36) [reply .claude/rules/tests.md:55]: Fixed in c4ca3d2: added the missing blank line before the heading, matching every other ## heading in the file.

File: ai/claude/rules/tests.md [hunk 3]
```diff
@@ -50,4 +50,13 @@ no explanatory comment: `test_renders_markdown_when_answer_present`,
...
+rest. No comments that restate the code, and no phase labels.
+## One behavior, one level
+
+Every behavior is tested at exactly one level:
+
+- **Unit** covers the logic you wrote: branches, transformations, computed properties.
+- **Integration** covers only the wiring unit tests cannot reach: real API contracts, serialization boundaries, external services.
+- **Evals** cover output quality against labeled data, never correctness already asserted below.
+
+Before writing a test, check whether a lower level already asserts the same behavior. If it does, do not write the test. When a higher-level test duplicates a lower-level assertion, delete the duplicate.
```
[the remark this answers is not in the document]
Comment (bertini36) [reply ai/claude/rules/tests.md]: Fixed in c4ca3d2: wrapped the new bullets and paragraph to ~80-86 chars, matching the rest of the file.

File: ai/claude/rules/performance.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+- Never query inside a loop. Fix N+1 at the queryset origin with `select_related` (FK/OneToOne) or `prefetch_related` (M2M/reverse FK).
+- Compute in the database with `annotate`/`aggregate` instead of looping over model instances in Python.
+- Batch writes: `bulk_create`, `bulk_update`, queryset `update()`. Never `.save()` in a loop.
+- Fetch only what you need in hot paths: `values()`, `values_list()`, or `only()`.
+
+## Tests as query guards
+
+- Any test covering a list endpoint or a serializer with relations asserts the query count with the `django_assert_num_queries` fixture, so a future N+1 fails a test instead of production.
+
```
[the remark this answers is not in the document]
Comment (bertini36) [reply ai/claude/rules/performance.md:27]: Fixed in 7f650eb: labeled the `django_assert_num_queries` bullet as Django-specific and added a stack-agnostic alternative for other ORMs.
